### PR TITLE
1956 angular tests stepper

### PR DESCRIPTION
**Pull request by daniilkopylov**

### CODE REVIEW

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import static io.github.com.pages.sections.StepperSection.linearStepper;
+import static io.github.com.pages.sections.StepperSection.optionalStepButton;
+import static io.github.com.pages.sections.StepperSection.optionalStepper;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class StepperTests extends TestsInit {
+    private static final String ENABLE_LINEAR = "Enable linear mode";
+    private static final String DISABLE_LINEAR = "Disable linear mode";
+    private static final String ENABLE_OPTIONAL = "Enable optional steps";
+    private static final String DISABLE_OPTIONAL = "Disable optional steps";
```
Comment (mf-code) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:19]: Codacy found an issue: [Avoid unused private fields such as 'DISABLE_OPTIONAL'.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034)

**Comment by mf-code** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/Stepper.java  1
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/StepperAssert.java  1
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java  2
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/StepperUnitTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034&bid=20661876)

**Comment by mf-code** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java  2
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/StepperUnitTests.java  1
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/StepperAssert.java  1
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/Stepper.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034&bid=20661876)

**Comment by mf-code** [comment]
[6317] - cYg0gpBrCemH

**Comment by mf-code** [comment]
[6317] - xSyxSqCBVEzd

**Comment by mf-code** [comment]
[6317] - MrvhoONszCRf

**Comment by mf-code** [comment]
[6317] - imXanJdiPJnT

**Comment by mf-code** [comment]
[6317] - 5TQE09xcQtgi

**Comment by mf-code** [comment]
[6317] - FJqkFIzKQ6ZF

**Comment by mf-code** [comment]
[6317] - EcMC5OoZDd8H

**Comment by mf-code** [comment]
[6317] - 8hu7d18aXyWx

**Comment by mf-code** [comment]
[6317] - 0OVabEC1r3x7

**Comment by mf-code** [comment]
[6317] - vZZc0ozKcMYz

**Comment by mf-code** [comment]
[6317] - 0aU1gxXnzdl1

**Comment by mf-code** [comment]
[6317] - LZcx5bpqt6PE

**Comment by mf-code** [comment]
[6317] - qoqnUIlVaLhn

**Comment by mf-code** [comment]
[6317] - U6K8LBlTrmNu

**Comment by mf-code** [comment]
[6318] - 6pAZAyyHwkpH

**Comment by mf-code** [comment]
[6318] - yObHx8TkRlMu

**Comment by mf-code** [comment]
[6318] - 6OsYLQPNF6VC

**Comment by mf-code** [comment]
[6318] - 4Qt3aTHXPXFy

**Comment by mf-code** [comment]
[6318] - KQ8lsoVJ3AFf

**Comment by mf-code** [comment]
[6318] - mbeVQRBVL253

**Comment by mf-code** [comment]
[6318] - VvEksnU11842

**Comment by mf-code** [comment]
[6318] - qwNBkeuuLJtP

**Comment by mf-code** [comment]
[6318] - T3HcbJ2l4hE8

**Comment by mf-code** [comment]
[6318] - uMbbVwUxiNsO

**Comment by mf-code** [comment]
[6318] - BUKWm7ubr0a6

**Comment by mf-code** [comment]
[6318] - EDadSaPHqDdg

**Comment by mf-code** [comment]
[6318] - NCJBdLLe2swr

**Comment by mf-code** [comment]
[6318] - hvULnDnoxUkf

**Comment by mf-code** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/StepperUnitTests.java  1
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/Stepper.java  1
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java  2
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/StepperAssert.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034&bid=20661876)

**Comment by mf-code** [comment]
[6319] - RscUAyjmWAlk

**Comment by mf-code** [comment]
[6319] - bs9cNRqlCqYj

**Comment by mf-code** [comment]
[6319] - TZxEEussbMZf

**Comment by mf-code** [comment]
[6319] - bBrgtt2DkBGG

**Comment by mf-code** [comment]
[6319] - 65qO0obAEYoC

**Comment by mf-code** [comment]
[6319] - GpBBgJ0I7e2G

**Comment by mf-code** [comment]
[6319] - PIQqRJBS7nYK

**Comment by mf-code** [comment]
[6319] - 4zSrJVM5B1h7

**Comment by mf-code** [comment]
[6319] - GkLl9n8JL466

**Comment by mf-code** [comment]
[6319] - TN2OIE1ntRpN

**Comment by mf-code** [comment]
[6319] - EGKv8xHMVJAe

**Comment by mf-code** [comment]
[6319] - F8otDZsgmwx1

**Comment by mf-code** [comment]
[6319] - 0pC1kHTHKpko

**Comment by mf-code** [comment]
[6319] - MqMZzOkkSl3M

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        linearStepper.show();
+        linearStepper.is().displayed();
+        optionalStepper.is().displayed();
+
+    }
+
+    @Test
+    public void basicStepperTest() {
+        linearStepper.show();
+        if (linearModeButton.getText().equals(DISABLE_LINEAR))
```
Comment (vsevolod-adrianov) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:37]: try use:        assertTrue(!linearModeButton.getText().equals(DISABLE_LINEAR));

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        linearStepper.gotoStep("2");
+        linearStepper.is().active("2");
+        linearStepper.gotoStep("1");
+        linearStepper.is().active("1").and().notActive("2");
+    }
+
+    @Test
+    public void basicLinearStepperTest() {
+        linearStepper.show();
+        if (linearModeButton.getText().equals(ENABLE_LINEAR))
```
Comment (vsevolod-adrianov) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:49]: assertTrue(linearModeButton.getText().equals(ENABLE_LINEAR));

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+            linearModeButton.click();
+        linearStepper.is().active("1");
+        linearStepper.gotoStep("2");
+        linearStepper.is().active("1").and().notActive("2");
+    }
+
+    @Test
+    public void basicOptionalStepperTest() {
+        optionalStepper.show();
+        if (optionalStepButton.getText().equals(ENABLE_OPTIONAL))
```
Comment (vsevolod-adrianov) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:59]: assertTrue(optionalStepButton.getText().equals(ENABLE_OPTIONAL));

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java (hunk 2 above)
Comment (vsevolod-adrianov) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:37]: remove it

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        linearStepper.is().displayed();
+        optionalStepper.is().displayed();
+
+    }
+
+    @Test
+    public void basicStepperTest() {
+        linearStepper.show();
+        if (linearModeButton.getText().equals(DISABLE_LINEAR))
+            linearModeButton.click();
```
Comment (vsevolod-adrianov) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:38]: remove it

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+import static io.github.com.pages.sections.StepperSection.linearModeButton;
+import static io.github.com.pages.sections.StepperSection.linearStepper;
+import static io.github.com.pages.sections.StepperSection.optionalStepButton;
+import static io.github.com.pages.sections.StepperSection.optionalStepper;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class StepperTests extends TestsInit {
+    private static final String ENABLE_LINEAR = "Enable linear mode";
+    private static final String DISABLE_LINEAR = "Disable linear mode";
+    private static final String ENABLE_OPTIONAL = "Enable optional steps";
```
Comment (mf-code) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:17]: Codacy found an issue: [Avoid unused private fields such as 'ENABLE_OPTIONAL'.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034)

**Comment by mf-code** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/StepperAssert.java  1
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/StepperUnitTests.java  1
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/Stepper.java  1
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034&bid=20661876)

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.sections.StepperSection.linearModeButton;
+import static io.github.com.pages.sections.StepperSection.linearStepper;
+import static io.github.com.pages.sections.StepperSection.optionalStepButton;
+import static io.github.com.pages.sections.StepperSection.optionalStepper;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class StepperTests extends TestsInit {
+    private static final String ENABLE_LINEAR = "Enable linear mode";
+    private static final String DISABLE_LINEAR = "Disable linear mode";
```
Comment (mf-code) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:16]: Codacy found an issue: [Avoid unused private fields such as 'DISABLE_LINEAR'.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034)

File: jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+
+import static io.github.com.StaticSite.angularPage;
+import static io.github.com.pages.sections.StepperSection.linearModeButton;
+import static io.github.com.pages.sections.StepperSection.linearStepper;
+import static io.github.com.pages.sections.StepperSection.optionalStepButton;
+import static io.github.com.pages.sections.StepperSection.optionalStepper;
+import static io.github.epam.site.steps.States.shouldBeLoggedIn;
+
+public class StepperTests extends TestsInit {
+    private static final String ENABLE_LINEAR = "Enable linear mode";
```
Comment (mf-code) [inline jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java:15]: Codacy found an issue: [Avoid unused private fields such as 'ENABLE_LINEAR'.](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034)

**Comment by mf-code** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/asserts/StepperAssert.java  1
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/unit/StepperUnitTests.java  1
- jdi-light-angular/src/main/java/com/epam/jdi/light/angular/elements/complex/Stepper.java  1
- jdi-light-angular-tests/src/test/java/io/github/epam/angular/tests/elements/complex/StepperTests.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/jdi-testing/jdi-light/pullRequest?prid=6367034&bid=20661876)

**Comment by mf-code** [comment]
[6323] - m1JBLiOmro1u

**Comment by mf-code** [comment]
[6323] - CJv6b9xvTdul

**Comment by mf-code** [comment]
[6323] - EM823KmLbObj

**Comment by mf-code** [comment]
[6323] - aduhNLmdq98k

**Comment by mf-code** [comment]
[6323] - dqVbRCbPjMKH

**Comment by mf-code** [comment]
[6323] - noEQd24jBIfy

**Comment by mf-code** [comment]
[6323] - 9kRNjAKoAtBV

**Comment by mf-code** [comment]
[6323] - 0NJG9xbHCUyw

**Comment by mf-code** [comment]
[6323] - NYN18jUpnBJA

**Comment by mf-code** [comment]
[6323] - awEBY9pEXG8s

**Comment by mf-code** [comment]
[6323] - 2rchG25wjvyF

**Comment by mf-code** [comment]
[6323] - vNTy9WLjOnXv

**Comment by mf-code** [comment]
[6323] - ufGoyAFLGQEi

**Comment by mf-code** [comment]
[6323] - hi8OTRxBEyQH

**Comment by mf-code** [comment]
[6324] - hw1GcNeNaQvz

**Comment by mf-code** [comment]
[6324] - r4PsSAdF29IC

**Comment by mf-code** [comment]
[6324] - e3Vl4VSY9DQq

**Comment by mf-code** [comment]
[6324] - 3RaFcks7kpoT

**Comment by mf-code** [comment]
[6324] - pNcOjn5SgSkv

**Comment by mf-code** [comment]
[6324] - pv62SJt3l8Tr

**Comment by mf-code** [comment]
[6324] - nRqT3MIOeWdr

**Comment by mf-code** [comment]
[6324] - aKKTWbVKJTB2

**Comment by mf-code** [comment]
[6324] - guKSaqEuodAn

**Comment by mf-code** [comment]
[6324] - 1IiCyyXAzEGr

**Comment by mf-code** [comment]
[6324] - uFFMvVkvTkBv

**Comment by mf-code** [comment]
[6324] - bMrfdyOHOZaQ

**Comment by mf-code** [comment]
[6324] - O7HJxl9BwByy

**Comment by mf-code** [comment]
[6324] - Ka95iQ9CdrKB

**Comment by mf-code** [comment]
[6324] - oQEFVgvzGmeL

**Comment by mf-code** [comment]
[6324] - 5Gyx1VRx8s8a

**Comment by mf-code** [comment]
[6324] - EpVoR3Rqyduq

**Comment by mf-code** [comment]
[6324] - ZxYp8AFvbSrr

**Comment by mf-code** [comment]
[6324] - hNp3rt27N4aJ

**Comment by mf-code** [comment]
[6324] - va91FufEc6IJ

**Comment by mf-code** [comment]
[6324] - 9Y6QrY6F8uNm

**Comment by mf-code** [comment]
[6324] - BcZTbOg1zQ2x

**Comment by mf-code** [comment]
[6324] - h6AQZ2mSuvTu

**Comment by mf-code** [comment]
[6324] - icWzQHR8Lps4

**Comment by mf-code** [comment]
[6324] - QRPnIFnQN8Mn

**Comment by mf-code** [comment]
[6324] - oIZXw8vNW2PS

**Comment by mf-code** [comment]
[6324] - asm6QsKG9I2c

**Comment by mf-code** [comment]
[6324] - DaLOIOmKA1nP

**Review by AlexSofft — APPROVED** [review]

**Comment by mf-code** [comment]
[6324] - 8GdVp64NjQY6

**Comment by mf-code** [comment]
[6324] - mNKtZxSZxLlw

**Comment by mf-code** [comment]
[6324] - a4sGikT2YT9H

**Comment by mf-code** [comment]
[6324] - bSpWSzitLaru

**Comment by mf-code** [comment]
[6324] - UPEtFOdhWwkD

**Comment by mf-code** [comment]
[6324] - KsQuy3l3A7kv

**Comment by mf-code** [comment]
[6324] - xS1w8Z3huU1m

**Comment by mf-code** [comment]
[6324] - SoTshxOb4Ok9

**Comment by mf-code** [comment]
[6324] - LVhP428xZS1a

**Comment by mf-code** [comment]
[6324] - alZAP1BFGKxa

**Comment by mf-code** [comment]
[6324] - J7721SmbjZDU

**Comment by mf-code** [comment]
[6324] - CyjaXffkngGy

**Comment by mf-code** [comment]
[6324] - 9YgZFIfq6iO4

**Comment by mf-code** [comment]
[6324] - UbiRLEeRrXVf

**Comment by mf-code** [comment]
[6324] - lYXSuCKPjofA

**Comment by RomanIovlev** [comment]
Merged to pull_request_for_Roman

**Pull request closed without merging**